frontend/leaderboard/src/Sheets.re

```diff
@@ -0,0 +1,93 @@
+open SheetsBinding;
+
+let clientId = "";
+let clientSecret = "";
+let redirectURI = "";
+let tokenPath = "token.json";
+let scopes = [|"https://www.googleapis.com/auth/spreadsheets.readonly"|];
+
+let readAndParseToken = () => {
+  switch (Node.Fs.readFileAsUtf8Sync(tokenPath)) {
+  | token => Some(Js.Json.parseExn(token))
+  | exception (Js.Exn.Error(_)) => None
+  };
+};
+
+let requestURICode = (client, cb) => {
+  let authUrlConfig = {access_type: "offline", scope: scopes};
+  let authUrl = generateAuthUrl(~client, ~authUrlConfig);
+  Js.log("Authorize this app by visiting this url: " ++ authUrl);
+
+  let readLine =
+    createInterface(interfaceOptions(~input=[%raw "process.stdin"]));
+
+  question(
+    readLine,
+    "Enter the code from that page here: ",
+    code => {
+      switch (code) {
+      | "" => cb(Error(code))
```

Comment:
Maybe this should be an actual error message like `Error("No code entered")` or something